mern/server/interactor/QA_interactor.js

```diff
@@ -0,0 +1,37 @@
+import QA_DAO from "../DAO/QA_dao.js";
+//Interactor calling the QA DAO functions after properly checking the inputs
+export default class QA_Interactor {
+  //Empty constructor
+  constructor(props) {
+    super(props);
+  }
+
+  //Gets the QA by ID
+  getQA(req, res) {
+    QA_DAO.getQAByID(req, res);
+  }
+
+  //Creates a new QA
+  createQA(req, res) {
+    QA_DAO.createQA(req, res);
+  }
+
+  //Updates QA if new QA object is valid and QA exists in collection
+  updateQA(req, res) {
+    const qa = QA_DAO.QAExists(req, res);
+    if (!qa) {
+      res.status(400, "QA not Found");
+    }
+    if (req.body.question && req.body.intents && req.body.question_included) {
+      qa.question = question;
```

Comment:
This needs to be changed so that if the test for req.body.hasOwnProperty does not pass, a res.status(400) is sent immediately without calling the update.

This is because in update, you are setting res.status which is set in here already. It can't be changed again afterwards, throws error.

Then only if it passes this check, you update the values of the transcript.